build_system/tdl-config.cmake

```diff
@@ -152,6 +152,17 @@ else ()
     tdl_config_print ("Optional dependency:        libexecinfo not found.")
 endif ()
 
+# ----------------------------------------------------------------------------
+# Set C++ Standard
+# ----------------------------------------------------------------------------
+
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    tdl_config_print("No C++ standard set via CMAKE_CXX_STANDARD: setting it to C++17")
+    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
```

Comment:
Can you still set it to `c++20` via command line?
I remember that in the app-template, it wouldn't use anything but `-std=c++17`, regardless of what was passed via command line...

Reply:
Good question, I will check!

Reply:
So it depends how it is being passed on the command line.
If you call `cmake ../test -DCMAKE_CXX_STANDARD=20` it will not reset it.
But if you call `cmake ../test -DCMAKE_CXX_FLAGS="-std=c++20"` it will set it to c++17 (It will actually set both on the command line, but at the end it will use the c++17 flag to compile the file)

I think using -DCMAKE_CXX_FLAGS to pass the c++ standard is very common.
But also it does bypass any build system functionality and shouldn't be used. There is a reason why -DCMAKE_CXX_STANDARD exists.
Not sure about this. How does seqan3 does it?

Reply:
We do it via `CXX_FLAGS`; back then there was no `-DCMAKE_CXX_STANDARD=20` ([introduced with 3.12](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html)).
A main reason for this is (I think) that we do automatic detection, i.e. we use either `-std=c++17 -fconcepts`, `-std=c++2a`, or `-std=c++20`. This is quite user-friendly, but as you mentioned `CMAKE_CXX_STANDARD` gets appended and hence the standard set in `CMAKE_CXX_FLAGS` is ignored.

I guess an alternative would be to use (untested)
```cmake
include (CheckCXXCompilerFlag)

check_cxx_compiler_flag ("-std=c++17" TDL_HAS_CPP17)
check_cxx_compiler_flag ("-std=c++2a" TDL_HAS_CPP2A)
check_cxx_compiler_flag ("-std=c++20" TDL_HAS_CPP20)

if (TDL_HAS_CPP20)
    set (CMAKE_CXX_FLAGS "-std=c++20 ${CMAKE_CXX_FLAGS}")
elseif (TDL_HAS_CPP2A)
    set (CMAKE_CXX_FLAGS "-std=c++2a ${CMAKE_CXX_FLAGS}")
elseif (TDL_HAS_CPP17)
    set (CMAKE_CXX_FLAGS "-std=c++17 ${CMAKE_CXX_FLAGS}")
else ()
    message (FATAL_ERROR "TDL needs at least CPP17.")
endif ()
```

Prepending the flags should cover the case that you pass a flag via `CMAKE_CXX_FLAGS`, since `CMAKE_CXX_FLAGS` will take precedence. 
I haven't looked into how SeqAn3 does it (I think it checks for set flags before checking support); but above version is probably good enough.

Reply:
Side Note: This part is actually not part of this PR, but part of #5, I will also fix it there according to your suggestion.